assets/js/data.coffee

```diff
@@ -268,20 +338,26 @@ class Data
 
     parent = do row.getParent
     children = @getSiblings row
-    i = @indexOf row
-    children.splice i, 1
+    ci = @indexOf row
+    children.splice ci, 1
+    parents = @getParents row
+    pi = _.findIndex parents, (par) ->
+        par == parent.id
+    parents.splice pi, 1
 
     @setChildren parent.id, children
+    @store.setParents row.id, parents
 
     return {
       parent: parent
-      index: i
+      index: ci
     }
 
   # attaches a detached child to a parent
   # the child should not have a parent already
   attachChild: (row, child, index = -1) ->
-    children = @attachChildren row, [child], index
+    @attachChildren row, [child], index
+    return child
```

Comment:
Nope, it doesn't, but I still think it's fine to return--less confusing than returning [child], which is the default otherwise.